packages/core/src/singleton/Interactions/types.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+interface LexV1BotConfig {
+	alias: string;
+	region: string;
+}
+
+interface LexV2BotConfig {
+	botId: string;
+	aliasId: string;
+	localeId: string;
+	region: string;
+}
+
+type InteractionsLexV1Config = {
+	LexV1?: Record<string, LexV1BotConfig>;
+};
+
+type InteractionsLexV2Config = {
+	LexV2?: Record<string, LexV2BotConfig>;
+};
+
+export type InteractionsConfig = InteractionsLexV1Config &
+	InteractionsLexV2Config;
```

Comment:
I think `InteractionsConfig` would also allow `{}`. How about something like this. 
```suggestion
type InteractionsLexV1Config = {
	LexV1: Record<string, LexV1BotConfig>;
};

type InteractionsLexV2Config = {
	LexV2: Record<string, LexV2BotConfig>;
};

export type InteractionsConfig = 
  | InteractionsLexV1Config
	| InteractionsLexV2Config
	| InteractionsLexV1Config & InteractionsLexV2Config
```

Reply:
For typescript intersection type, my understanding is 
```
{A?} & {B?} ≅ ({} | {A}) & ({} | {B}) ≅ {} | {A} | {B} | {A, B} 
```
it already contains all the cases you mentioned above.

Reply:
yep makes sense, but i'm wondering if we should avoid `{}`

Reply:
As we are doing merging at category configuration level, running `Amplify.configure({Interactions:{}})` will have no effect on the configuration.